src/classes/CW_AdminControllerTest.cls

```diff
@@ -0,0 +1,199 @@
+		Set<Id> audienceGroupMemberIds = new Set<Id>();
+		for (CW_Audience_Group_Member__c agm :audienceGroupMembers) {
+			system.assertEquals(0, agm.Lifetime_NGR__c);
+			system.assertEquals(0, agm.Lifetime_Wager__c);
+			audienceGroupMemberIds.add(agm.Id);
+		}
+		
+    	Test.startTest();
+        Pagereference pageRef = Page.CW_Admin;
+        Test.setCurrentPage(pageRef);
+        
+        CW_AdminController controller = new CW_AdminController();
+        controller.updateAudienceGroupMembers();
+        Test.stopTest();
+        
+        // get updated audience group members 
+		// (batch job should have stamped KPI's from contact record on group member record where run batch date was inside the eval dates of a schedule)
+		List<CW_Audience_Group_Member__c> updatedAudienceGroupMembers = 
+			[select Id, Lifetime_NGR__c, Lifetime_Wager__c from CW_Audience_Group_Member__c where Id in :audienceGroupMemberIds];
+			
+		for (CW_Audience_Group_Member__c agm :updatedAudienceGroupMembers) {
+			if (agm.Id == audienceGroupMembers[0].Id) {
+				// run batch date was inside period of schedule; KPI's have been updated from Contact to assocated Audience Group Member record
+				system.assertEquals(contacts[0].Lifetime_NGR__c, agm.Lifetime_NGR__c);
+				system.assertEquals(contacts[0].Lifetime_Wager__c, agm.Lifetime_Wager__c);
+			} 
+		}
+    }
+    
+    @isTest static void testAdminControllerAddAndRemoveScheduleBatchUpdateAudienceGroupMembers() {
+    	// setup data
+		CW_CampaignListSetting__c batchRunDateRecord = new CW_CampaignListSetting__c();
+		batchRunDateRecord.Name = ConstantUtil.BATCH_RUN_DATE;
+		batchRunDateRecord.Date_Value__c = Date.today();
+		insert batchRunDateRecord;
+		
+		// Instantiate the batch class
+		CW_BatchUpdateAudienceGroupMember myBatch = new CW_BatchUpdateAudienceGroupMember();
+ 
+		// Instantiate the scheduler
+		BatchScheduler scheduler = new BatchScheduler();
+ 
+		// Assign the batch class to the variable within the scheduler
+		scheduler.batchClass = myBatch;
+ 
+		// Get Schedule string - run every day at 1am (01:00)
+		String sch = ConstantUtil.BATCH_APEX_SCHEDULE_UPDATE_AUDIENCE_GROUP_MEMBER;
+ 
+ 		Test.startTest();
+ 		Pagereference pageRef = Page.CW_Admin;
+        Test.setCurrentPage(pageRef);
+        
+        CW_AdminController controller = new CW_AdminController();
+        controller.addScheduledJobToUpdateAudienceGroupMembers();
+		Test.stopTest();
+		
+		// assert that batch job is scheduled
+		List<CronTrigger> cronJobs = [
+	    	select Id, CronJobDetailId, CronJobDetail.Name 
+	    	from CronTrigger where CronJobDetail.Name LIKE :ConstantUtil.BATCH_APEX_JOB_NAME_UPDATE_AUDIENCE_GROUP_MEMBER+'%'
+	    ];
+	    system.assertEquals(true, cronJobs.size()==1);
+	    
+	    // remove schedule
+	    controller.removeScheduledJobToUpdateAudienceGroupMembers();
+	    
+	    // assert that batch job is scheduled
+		cronJobs = [
+	    	select Id, CronJobDetailId, CronJobDetail.Name 
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.